### ISSUE TITLE
Program usually hangs during the forward pass.

### ISSUE THREAD

**Issue by awasthiabhijeet** [pytorch/xla#2163]
## ❓ Questions and Help
Hi, I am trying to deploy [Jasper](https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/SpeechRecognition/Jasper), a speech recognition model solely composed of convolutions, batch normalization, dropout, and ReLU. The network has residual connections.

The audio processing module runs on the CPU, while the neural modules are shifted to TPUs.
My program usually hangs during the inference, as soon as the forward pass of the model is called.
The forward pass did work for a few times without any problem. So, I am not sure what's the reason behind this.

What would be the recommended procedure to debug such a problem?

**Comment by dlibenzi** [comment]
I would start by creating a Colab based on the following template, so we can reproduce it and debug it:

https://github.com/pytorch/xla/blob/master/contrib/colab/issue-report.ipynb

**Comment by awasthiabhijeet** [comment]
Okay, thanks for the reply. I will share a Colab soon.

Meanwhile, I guess, using [CTCLoss](https://pytorch.org/docs/master/generated/torch.nn.CTCLoss.html) may be a bottleneck.
Is CTCLoss currently supported?

**Comment by dlibenzi** [comment]
Currently not supported.
